pretyping/evarconv.ml

```diff
@@ -176,6 +176,10 @@ let check_conv_record env sigma (t1,sk1) (t2,sk2) =
         let s = ESorts.kind sigma s in
 	lookup_canonical_conversion
 	  (proji, Sort_cs (family_of_sort s)),[]
+      | Proj (p, c) ->
+        let c2 = Globnames.ConstRef (Projection.constant p) in
+        let sk2 = Stack.append_app [|c|] sk2 in
+        lookup_canonical_conversion (proji, Const_cs c2), sk2
       | _ ->
```

Comment:
Isn't it the kind of pattern that we are supposed to get rid of in the code? So as not to forget, let's say, to adapt the code of canonical structures whenever primitive projections are added?
BTW, isn't there an OCaml warning which could be activated to prevent this kind of style?